.github/workflows/update-tektoncd-task-versions.yml

```diff
@@ -0,0 +1,66 @@
+name: Update Tekton Task Versions
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs daily at midnight
+  workflow_dispatch:  # Allows manual trigger
+
+jobs:
+  update-task-versions:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'tektoncd' # do not run this elsewhere
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout Operator Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Clone TektonCD Catalog (p branch)
+        run: |
+          git clone --branch p https://github.com/openshift-pipelines/tektoncd-catalog.git ../tektoncd-catalog
+
+      - name: Update Task Versions
+        id: update
+        run: |
+          chmod +x hack/openshift/update-tekton-task-versions.sh
+          ./hack/openshift/update-tekton-task-versions.sh \
```

Comment:
Is this script openshift specific? If so, we should probably indicate this in the name.